hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -824,18 +834,19 @@ private void handleSuccessfulAuth(TcpClientConnection connection, ClientAuthenti
                 logger.fine("Checking the cluster: " + newClusterId + ", current cluster: " + this.clusterId);
             }
 
-            boolean initialConnection = activeConnections.isEmpty();
-            boolean changedCluster = initialConnection && this.clusterId != null && !newClusterId.equals(this.clusterId);
-            if (changedCluster) {
+            boolean connectionsEmpty = activeConnections.isEmpty();
+            boolean clusterIdChanged = this.clusterId != null && !newClusterId.equals(this.clusterId);
+            if (clusterIdChanged) {
+                checkClientStateOnClusterIdChange(connection);
```

Comment:
you can pass parameter `connectionsEmpty ` to this method.

Reply:
isEmpty is not an expensive call. I would rather leave it this way.